src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx

```diff
@@ -0,0 +1,1115 @@
+import { ChartValuesSelect } from '../../../charts/util/ChartValueSelect'
+import { ConfirmationDialog, DeleteDialog, DetailsProgressing, Progressing, Select, showError } from '../../../common'
+import {
+    ChartEnvironmentSelectorType,
+    ChartRepoSelectorType,
+    ChartVersionSelectorType,
+    ChartValuesSelectorType,
+    ChartVersionValuesSelectorType,
+    ChartValuesEditorType,
+    ChartRepoDetailsType,
+    ChartProjectSelectorType,
+    ChartGroupOptionType,
+    ChartValuesDiffOptionType,
+    ChartRepoOtions,
+} from './ChartValuesView.type'
+import { getChartsByKeyword, getChartValues } from '../../../charts/charts.service'
+import CodeEditor from '../../../CodeEditor/CodeEditor'
+import { NavLink } from 'react-router-dom'
+import { Moment12HourFormat, URLS } from '../../../../config'
+import Tippy from '@tippyjs/react'
+import { MarkDown } from '../../../charts/discoverChartDetail/DiscoverChartDetails'
+import moment from 'moment'
+import { getDeploymentManifestDetails } from '../../chartDeploymentHistory/chartDeploymentHistory.service'
+import YAML from 'yaml'
+import EmptyState from '../../../EmptyState/EmptyState'
+import { ChartValuesViewAction } from './ChartValuesView.reducer'
+
+export const ChartEnvironmentSelector = ({
+    isExternal,
+    isDeployChartView,
+    installedAppInfo,
+    releaseInfo,
+    isUpdate,
+    selectedEnvironment,
+    handleEnvironmentSelection,
+    environments,
+    invalidaEnvironment,
+}: ChartEnvironmentSelectorType): JSX.Element => {
+    return !isDeployChartView ? (
+        <div className="chart-values__environment-container mb-12">
+            <h2 className="chart-values__environment-label fs-13 fw-4 lh-20 cn-7">Environment</h2>
+            {isExternal ? (
+                <span className="chart-values__environment fs-13 fw-6 lh-20 cn-9">
+                    {installedAppInfo
+                        ? installedAppInfo.environmentName
+                        : releaseInfo.deployedAppDetail.environmentDetail.clusterName +
+                          '__' +
+                          releaseInfo.deployedAppDetail.environmentDetail.namespace}
+                </span>
+            ) : (
+                <span className="chart-values__environment fs-13 fw-6 lh-20 cn-9">{selectedEnvironment.label}</span>
+            )}
+        </div>
+    ) : (
+        <div className="form__row form__row--w-100 fw-4">
+            <span className="form__label required-field">Deploy to environment</span>
+            <ReactSelect
+                components={{
+                    IndicatorSeparator: null,
+                    DropdownIndicator,
+                    Option,
+                }}
+                isDisabled={!!isUpdate}
+                classNamePrefix="values-environment-select"
+                placeholder="Select Environment"
+                value={selectedEnvironment}
+                styles={{
+                    menuList: (base) => ({
```

Comment:
Can we make these styles common so that it can be reused.

Reply:
Done.